src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DynamicInterfaceCastableImplementation.cs

```diff
@@ -0,0 +1,127 @@
+                new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.MembersDeclaredOnImplementationTypeMustBeSealedDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            DynamicInterfaceCastableImplementationUnsupported,
+            InterfaceMembersMissingImplementation,
+            MembersDeclaredOnImplementationTypeMustBeSealed);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDynamicInterfaceCastableImplementationAttribute, out INamedTypeSymbol? dynamicInterfaceCastableImplementationAttribute))
+                {
+                    context.RegisterSymbolAction(context => AnalyzeType(context, dynamicInterfaceCastableImplementationAttribute), SymbolKind.NamedType);
+                }
+            });
+        }
+
+        private static void AnalyzeType(SymbolAnalysisContext context, INamedTypeSymbol dynamicInterfaceCastableImplementationAttribute)
+        {
+            INamedTypeSymbol targetType = (INamedTypeSymbol)context.Symbol;
+
+            if (targetType.TypeKind != TypeKind.Interface)
+            {
+                return;
+            }
+
+            if (!targetType.HasAttribute(dynamicInterfaceCastableImplementationAttribute))
+            {
+                return;
+            }
+
+            // Default Interface Methods are required to provide an IDynamicInterfaceCastable implementation type.
+            // Since Visual Basic does not support DIMs, an implementation type cannot be correctly provided in VB.
+            if (context.Compilation.Language == LanguageNames.VisualBasic)
+            {
+                context.ReportDiagnostic(targetType.CreateDiagnostic(DynamicInterfaceCastableImplementationUnsupported));
+                return;
+            }
+
+            bool missingMethodImplementations = false;
+            foreach (var iface in targetType.AllInterfaces)
+            {
+                foreach (var member in iface.GetMembers())
+                {
+                    if (!member.IsStatic
+                        && context.Compilation.IsSymbolAccessibleWithin(member, targetType)
+                        && targetType.FindImplementationForInterfaceMember(member) is null)
+                    {
+                        missingMethodImplementations = true;
+                        break;
```

Comment:
💡 This only breaks out of the inner loop. It looks like we could break outside the outer loop too for efficiency.